demo/installer/grub-arch/install.sh

```diff
@@ -72,9 +72,12 @@ create_demo_gpt_partition()
     # Find next available partition
     last_part=$(sgdisk -p $blk_dev | tail -n 1 | awk '{print $1}')
     demo_part=$(( $last_part + 1 ))
+    # check if we have an mmcblk device
+    blk_suffix=
+    echo ${blk_dev} | grep -q mmcblk && blk_suffix="p"
 
     # Create new partition
-    echo "Creating new demo partition ${blk_dev}$demo_part ..."
+    echo "Creating new demo partition ${blk_dev}$blk_suffix$demo_part ..."
```

Comment:
Sorry.  It's my fault.  It needs the suffix here.  BTW, I found it has issue in the statement of extrating `blk_dev`

    blk_dev=$(blkid | grep ONIE-BOOT | awk '{print $1}' |  sed -e 's/[1-9][0-9]*:.*$//' | sed -e 's/\([0-9]\)\(p\)/\1/' | head -n 1)

where `sed -e 's/\([0-9]\)\(p\)/\1/'` would not remove partition number correctly.  For instance:

    # blk_dev=/dev/mmcblk0p1
    # echo $blk_dev | sed -e 's/\([0-9]\)\(p\)/\1/'
    /dev/mmcblk01